lib/geoengineer/resources/aws/certificate_manager/aws_acmpca_certificate_authority.rb

```diff
@@ -0,0 +1,40 @@
+########################################################################
+# AwsAcmpcaCertificateAuthority is the +aws_acmpca_certificate_authority+
+# terraform resource,
+#
+# {https://www.terraform.io/docs/providers/aws/r/acmpca_certificate_authority.html}
+########################################################################
+class GeoEngineer::Resources::AwsAcmpcaCertificateAuthority < GeoEngineer::Resource
+  after :initialize, -> { _terraform_id -> { NullObject.maybe(remote_resource)._terraform_id } }
+
+  validate -> { validate_required_attributes([:type]) }
+  validate -> { validate_required_attributes([:tags]) }
+  validate -> { validate_required_attributes([:certificate_authority_configuration]) }
+  validate -> { validate_subresource_required_attributes(:certificate_authority_configuration, [:subject]) }
+  validate -> { validate_subresource_required_attributes(:certificate_authority_configuration, [:key_algorithm]) }
+  validate -> { validate_subresource_required_attributes(:certificate_authority_configuration, [:signing_algorithm]) }
+  validate -> { validate_subresource_required_attributes(:subject, [:common_name]) }
+
+  after :initialize, -> { _geo_id -> { [type, common_name].join("::") } }
+  def common_name
+    if !self.certificate_authority_configuration.nil? &&
+       !self.certificate_authority_configuration.subject.nil? &&
+       !self.certificate_authority_configuration.subject.common_name.nil?
+
+      return self.certificate_authority_configuration.subject.common_name
+    end
+    NullObject.new()
```

Comment:
Nit: I don't know if we have RuboCop configured on this repo, but including the `()` in Ruby method calls is optional.

Reply:
It is turned on, but that style is not being flagged at the moment.